packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -179,6 +181,126 @@ const Conversation = WebexPlugin.extend({
+    // property. we need jwk.k to correctly generate the HMAC
+
+    return this.webex.internal.encryption.unboundedStorage.get(parent.encryptionKeyUrl)
+      .then((keyString) => {
+        const key = JSON.parse(keyString);
+        // when we stringify this object, keys must be in this order to generate same HMAC as
+        // desktop clients
+        const formatjwk = {k: key.jwk.k, kid: key.jwk.kid, kty: key.jwk.kty};
+
+        const source = `${JSON.stringify(formatjwk)}${parent.id}${displayName}`;
+
+        const hmac = hex.stringify(hmacSHA256(source, parent.id));
+
+        return Promise.resolve(hmac);
+      });
+  },
+
+  /**
+   * @private
+   * send add or delete reaction to convo service
+   * @param {Object} conversation
+   * The payload to send a reaction
+   * @typedef {Object} ReactionPayload
+   * @property {Object} actor
+   * @property {string} actor.objectType
+   * @property {string} actor.id
+   * @property {string} objectType
+   * @property {string} verb will be either add' or 'delete'
+   * @property {Object} target
+   * @property {string} target.id
+   * @property {string} target.objectType
+   * @property {Object} object this will change on delete vs. add
+   * @property {string} object.id present in delete case
+   * @property {string} object.objectType 'activity' in delete case, 'reaction2' in add case
+   * @property {string} object.displayName must be 'celebrate', 'heart', 'thumbsup', 'smiley', 'haha', 'confused', 'sad'
+   * @property {string} object.hmac
+   * @param {ReactionPayload} reactionPayload
+   * @returns {Promise<Activity>}
+   */
+  sendReaction(conversation, reactionPayload) {
+    const url = this.getConvoUrl(conversation);
+    const convoWithUrl = Object.assign({}, conversation, {url});
+
+    if (!isObject(reactionPayload)) {
+      return Promise.reject(new Error('`object` must be an object'));
+    }
+
+    return this.prepare(reactionPayload, {
+      target: this.prepareConversation(convoWithUrl),
+      object: pick(reactionPayload, 'id', 'url', 'objectType')
+    })
+      .then((act) => this.submit(act));
+  },
+
+  /**
+   * delete a reaction
+   * @param {Object} conversation
+   * @param {Object} reactionId
+   * @returns {Promise<Activity>}
+   */
+  deleteReaction(conversation, reactionId) {
+    const deleteReactionPayload = {
+      actor: {objectType: 'person', id: this.webex.internal.device.userId},
+      object: {
+        id: reactionId,
+        objectType: 'activity'
+      },
+      objectType: 'activity',
+      target: {
+        id: conversation.id,
+        objectType: 'conversation'
+      },
+      verb: 'delete'
+    };
+
+    return this.sendReaction(conversation, deleteReactionPayload);
+  },
+
+  /**
+   * create a reaction
+   * @param {Object} conversation
+   * @param {Object} displayName must be 'celebrate', 'heart', 'thumbsup', 'smiley', 'haha', 'confused', 'sad'
+   * @param {Object} parent activity object from convo we are reacting to
```

Comment:
Can we just rename this to activity for consistency with the other methods in the plugin that takes an activity?

```suggestion
   * @param {Activity} activity object from convo we are reacting to
```